src/store/albums.js

```diff
@@ -44,13 +44,27 @@ export default {
         }
       }
     },
-    updateLabelOccurence(state, { oldID, newID }) {
+    removeArtistOccurence(state, { oldID }) {
+      const oldAlbums = state.albums;
+      state.albums = {};
+      for (let album in Object.values(oldAlbums)) {
+        const i = album.album_artists.findIndex((aa) => aa.artist_id === oldID);
+        if (i >= 0) {
+          album.album_artists.splice(i, 1);
+        }
+        state.albums[album.id] = album;
+      }
+    },
+    removeOrMergeLabelOccurence(state, { oldID, newID }) {
```

Comment:
I'm not sure about doing the remove and merge actions in the same function. There's some shared boilerplate, but I would still split these up. (Same for genre.)

Reply:
It felt like too much boilerplate to me, but sure.